<!DOCTYPE html>
<html lang="en-us">
  <head>
    <meta charset="UTF-8">
    <title>grinn R package</title>
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <link rel="stylesheet" type="text/css" href="stylesheets/normalize.css" media="screen">
    <link href='http://fonts.googleapis.com/css?family=Open+Sans:400,700' rel='stylesheet' type='text/css'>
    <link rel="stylesheet" type="text/css" href="stylesheets/stylesheet.css" media="screen">
    <link rel="stylesheet" type="text/css" href="stylesheets/github-light.css" media="screen">
  </head>
  <body>
    <section class="page-header">
      <img src="images/logo.png" alt="" width="450px">
      <a href="https://github.com/kwanjeeraw/grinn" class="btn">View on GitHub</a>
      <a href="https://github.com/kwanjeeraw/grinn/zipball/master" class="btn">Download .zip</a>
      <a href="https://github.com/kwanjeeraw/grinn/tarball/master" class="btn">Download .tar.gz</a>
    </section>
    <section class="main-content">
      <hr>
          <h2><a id="fetchCorrGrinnNetwork" class="anchor" href="#fetchCorrGrinnNetwork" aria-hidden="true"><span class="octicon octicon-link"></span></a>fetchCorrGrinnNetwork - compute a weighted correlation network and expand the network with information from grinn internal database</h2>
          <h3>Description</h3>
          <p>from input omics data e.g. normalized expression data or metabolomics data, it is a one step function to:<br>
            1. Compute a weighted correlation network of input omics data using WGCNA functions cor and corPvalueStudent. The correlation coefficients are continuous values
            between -1 (negative correlation) and 1 (positive correlation), with numbers close to 1 or -1, meaning very closely correlated. Then the correlation network is built by function fetchCorrNetwork.<br>
            2. Expand the correlation network using information from grinn internal database. The nodes of the correlation network are the keywords input to query the Grinn database.
            Grinn internal database contains the networks of the following types that can get expanded to: metabolite-protein, metabolite-protein-gene, metabolite-pathway, protein-gene, protein-pathway and gene-pathway,
            see also fetchGrinnNetwork.</p>
          <h3>Usage</h3>
          <p>fetchCorrGrinnNetwork(datNormX, datNormY, corrCoef, pval, method, returnAs, sourceTo, targetTo, filterSource)</p>
          <h3>Arguments</h3>
          <table>
            <tr><td>datNormX</td><td>data frame containing normalized, quantified omics data e.g. expression data, metabolite intensities. Columns correspond to entities e.g. genes, metabolites, and rows to samples e.g. normals, tumors. Require 'nodetype' at the first row to indicate the type of entities in each column. See below for details.</td></tr>
            <tr><td>datNormY</td><td>data frame containing normalized, quantified omics data e.g. expression data, metabolite intensities. Use the same format as datNormX or it can be NULL. See below for details.</td></tr>
            <tr><td>corrCoef</td><td>numerical value to define the minimum value of absolute correlation, from 0 to 1, to include edges in the output.</td></tr>
            <tr><td>pval</td><td>numerical value to define the maximum value of pvalues, to include edges in the output.</td></tr>
            <tr><td>method</td><td>string to define which correlation is to be used. It can be one of "pearson","kendall","spearman" (default), see cor.</td></tr>
            <tr><td>returnAs</td><td>string of output type. Specify the type of the returned network. It can be one of "tab","json","cytoscape", default is "tab". "cytoscape" is the format used in Cytoscape.js</td></tr>
            <tr><td>sourceTo</td><td>string of node type. It can be one of "metabolite","protein","gene","pathway". See below for details.</td></tr>
            <tr><td>targetTo</td><td>string of node type. It can be one of "metabolite","protein","gene","pathway". By default, it will expand to pathways, see below for details.</td></tr>
            <tr><td>filterSource</td><td>string or list of pathway databases. The argument is required, if sourceTo or targetTo = "pathway". The argument value can be any of "SMPDB","KEGG","REACTOME" or combination of them e.g. list("KEGG","REACTOME").</td></tr>
          </table>
          <h3>Details</h3>
          <p>datNormX and datNormY are matrices in which rows are samples and columns are entities. If datNormY is given, then the correlations between the columns of datNormX and the columns of datNormY are computed. In this case:
          <br>- The correlation network can be expand from datNormX entites to a specific node type, by providing a value to sourceTo
          <br>- The correlation network can be expand from datNormY entites to a specific node type, by providing a value to targetTo
          <br>Otherwise if datNormY is not given, the correlations of the columns of datNormX are computed. In this case:
          <br>- The correlation network can be expand from datNormX entites to a specific node type, by providing a value to targetTo and leave sourceTo = NULL.<br>
          The column names of both datNormX and datNormY are required to use grinn ids. convertToGrinnID is provided for id conversion, see convertToGrinnID.</p>
          <h3>Value</h3>
          <p>list of nodes and edges. The list is with the following componens: edges and nodes. Return empty list if found nothing</p>
          <h3>Examples</h3>
          <pre><code>
# Compute a correlation network of metabolites and expand to a grinn network of metabolite-protein
dummy <- rbind(nodetype=rep("metabolite"),t(mtcars))
colnames(dummy) <- c('X1.1','X27967','X371','X4.1',paste0('X',sample(400:22000, 28)))
result <- fetchCorrGrinnNetwork(datNormX=dummy, corrCoef=0.7, pval=1e-12, method="spearman", returnAs="tab", targetTo="protein")
library(igraph)
plot(graph.data.frame(result$edges[,1:2], directed=FALSE))

# Compute a correlation network of metabolites and proteins and expand to the grinn network of metabolite-pathway and protein-gene
dummyX <- rbind(nodetype=rep("metabolite"),t(mtcars)[,1:16])
colnames(dummyX) <- c('X1.1','X27967','X371','X4.1',paste0('X',sample(400:22000, 12)))
dummyY <- rbind(nodetype=rep("protein"),t(mtcars)[,17:32])
colnames(dummyY) <- c('P28845','P08235','Q08AG9','P80365',paste0('P',sample(10000:80000, 12)))
result <- fetchCorrGrinnNetwork(datNormX=dummyX, datNormY=dummyY, corrCoef=0.7, pval=1e-4, method="spearman", returnAs="json", sourceTo="pathway", targetTo="gene")
          </pre></code>
          <h3><a id="references" class="anchor" href="#references" aria-hidden="true"><span class="octicon octicon-link"></span></a>References</h3>
          <p>Correlation-based analyses apply methods from the following publications:</p>
          <ul>
            <li>Dudoit S, <em>et al.</em> Statistical methods for identifying differentially expressed genes in replicated cDNA microarray experiments, <a href="http://citeseerx.ist.psu.edu/viewdoc/summary?doi=10.1.1.117.9702" target="_blank">STATISTICA SINICA 2002;12:111</a>.</li>
            <li>Langfelder P, <em>et al.</em> WGCNA: an R package for weighted correlation network analysis. <a href="http://www.biomedcentral.com/1471-2105/9/559" target="_blank">BMC Bioinformatics 2008;9:559</a>.</li>
            <li>Langfelder P, <em>et al.</em> Tutorials for the WGCNA package, <a href="http://labs.genetics.ucla.edu/horvath/CoexpressionNetwork/Rpackages/WGCNA/Tutorials/index.html" target="_blank"> click here</a></li> 
          </ul>
          <p>Go to <a href="index.html">HOME</a> | <a href="document.html">Documentation</a> | <a href="fgrinn.html">fetchGrinnNetwork</a> | <a href="fcorrgrinn.html">fetchCorrGrinnNetwork</a> | <a href="fdiffcorr.html">fetchDiffCorrGrinnNetwork</a>
          | <a href="fmodule.html">fetchModuGrinnNetwork</a> | <a href="fgrinncorr.html">fetchGrinnCorrNetwork</a> | <a href="finput.html">input formats</a></p>
        <hr>
        <footer>
          &copy;KW UC Davis, 2015, Grinn is maintained by <a href="https://github.com/kwanjeeraw">kwanjeeraw</a><br>
          This page was generated by <a href="http://pages.github.com">GitHub Pages</a>. Tactile theme by <a href="https://twitter.com/jasonlong">Jason Long</a>.
        </footer>
      </section>
          <script type="text/javascript">
            var gaJsHost = (("https:" == document.location.protocol) ? "https://ssl." : "http://www.");
            document.write(unescape("%3Cscript src='" + gaJsHost + "google-analytics.com/ga.js' type='text/javascript'%3E%3C/script%3E"));
          </script>
          <script type="text/javascript">
            try {
              var pageTracker = _gat._getTracker("UA-61396514-2");
            pageTracker._trackPageview();
            } catch(err) {}
          </script>
  </body>
</html>
