<!DOCTYPE html>
<html lang="en-us">
  <head>
    <meta charset="UTF-8">
    <title>grinn R package</title>
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <link rel="stylesheet" type="text/css" href="stylesheets/normalize.css" media="screen">
    <link href='http://fonts.googleapis.com/css?family=Open+Sans:400,700' rel='stylesheet' type='text/css'>
    <link rel="stylesheet" type="text/css" href="stylesheets/stylesheet.css" media="screen">
    <link rel="stylesheet" type="text/css" href="stylesheets/github-light.css" media="screen">
  </head>
  <body>
    <section class="page-header">
      <img src="images/logo.png" alt="" width="450px">
      <a href="https://github.com/kwanjeeraw/grinn" class="btn">View on GitHub</a>
      <a href="https://github.com/kwanjeeraw/grinn/zipball/master" class="btn">Download .zip</a>
      <a href="https://github.com/kwanjeeraw/grinn/tarball/master" class="btn">Download .tar.gz</a>
    </section>
    <section class="main-content">
      <hr>
          <h2><a id="fetchWGCNAModule" class="anchor" href="#fetchWGCNAModule" aria-hidden="true"><span class="octicon octicon-link"></span></a>fetchWGCNAModule - Compute a network module correlated to a phenotypic feature</h2>
          <h3>Description</h3>
          <p>identify correlation between the input omics data e.g. normalized gene expression data, and phenotypic data e.g. weight.
          The function wraps around important aspects of WGCNA including blockwiseModules, cor, corPvalueStudent, labeledHeatmap. These aspects automatically perform correlation network construction, module detection, and display module-phenotype correlations.
          A module or the combination of modules can be selected from the heatmap of module-phenotype correlations for including in the network output, see more details below.</p>
          <h3>Usage</h3>
          <p>fetchWGCNAModule(datNorm, datPheno, sfPower, minModuleSize, threshold, returnAs)</p>
          <h3>Arguments</h3>
          <table>
            <tr><td>datNorm</td><td>data frame containing normalized, quantified omics data e.g. expression data, metabolite intensities. Columns correspond to entities e.g. genes, metabolites, and rows to samples e.g. normals, tumors.
            Require 'nodetype' at the first row to indicate the type of entities in each column.</td></tr>
            <tr><td>datPheno</td><td>data frame containing phenotypic data e.g. weight, age, insulin sensitivity. Columns correspond to phenotypes and rows to samples e.g. normals, tumors.</td></tr>
            <tr><td>sfPower</td><td>numerical value of soft-thresholding power for correlation network construction. It is automatically estimated using pickSoftThreshold, or it can be defined by users.</td></tr>
            <tr><td>minModuleSize</td><td>numerical value of minimum module size for module detection.</td></tr>
            <tr><td>threshold</td><td>numerical value to define the minimum value of similarity threshold, from 0 to 1, to include edges in the output.</td></tr>
            <tr><td>returnAs</td><td>string of output type. Specify the type of the returned network. It can be one of "tab","json","cytoscape", default is "tab". "cytoscape" is the format used in Cytoscape.js</td></tr>
          </table>   
          <h3>Value</h3>
          <p>list of nodes and edges. The list is with the following componens: edges and nodes. Return empty list if found nothing</p>
          <h3>Details</h3>
          <p>The function encapsulates several methods from WGCNA so that module-phenoty correlation analysis can be fasten. These methods include:
          <br>- pickSoftThreshold estimates soft-thresholding powers from scale free topology to build the correlation network.
          <br>- blockwiseModules automatically calculates a correlation network and detects modules. Modules are the areas of the network where nodes are densely connected based on their topological overlap measure, see WGCNA for more details. Each module is labeled by color. By using the color, a module or the combination of modules can be selected ("enter color to the terminal"), for including in the network output.
          <br>- Module-phenotype correlations and significances are calculated using WGCNA functions cor and corPvalueStudent. The correlation coefficients are continuous values between -1 (negative correlation) and 1 (positive correlation), with numbers close to 1 or -1, meaning very closely correlated.
          <br>- labeledHeatmap plots a heatmap of module-phenotype correlations. A row is a module and a column is a phynotype. Each cell presents the corresponding correlation and the pvalue inside parenthesis. Each cell is colored by correlation, red means positive and blue means negative correlation.
          <br>- exportNetworkToCytoscape exports a network for using in Cytoscape (http://cytoscape.org/). The selected module is exported as the network output in which an edge will be included if it similarity threshold above the cutoff, see threshold.</p>
          <h3>Examples</h3>
          <pre><code>
# Compute a correlation of metabolite module to phenotypic data
library(grinn)
data(dummy)
data(dummyPheno)
result <- fetchWGCNAModule(datNorm=dummy, datPheno=dummyPheno, minModuleSize=5, threshold=0.2)
# enter module color(s) seperate by space:yellow brown purple
library(igraph)
plot(graph.data.frame(result$edges[,1:2], directed=FALSE))
          </pre></code>
          <h3><a id="references" class="anchor" href="#references" aria-hidden="true"><span class="octicon octicon-link"></span></a>References</h3>
          <p>Correlation-based analyses apply methods from the following publications:</p>
          <ul>
            <li>Dudoit S, <em>et al.</em> Statistical methods for identifying differentially expressed genes in replicated cDNA microarray experiments, <a href="http://citeseerx.ist.psu.edu/viewdoc/summary?doi=10.1.1.117.9702" target="_blank">STATISTICA SINICA 2002;12:111</a>.</li>
            <li>Langfelder P, <em>et al.</em> WGCNA: an R package for weighted correlation network analysis. <a href="http://www.biomedcentral.com/1471-2105/9/559" target="_blank">BMC Bioinformatics 2008;9:559</a>.</li>
            <li>Langfelder P, <em>et al.</em> Tutorials for the WGCNA package, <a href="http://labs.genetics.ucla.edu/horvath/CoexpressionNetwork/Rpackages/WGCNA/Tutorials/index.html" target="_blank"> click here</a></li> 
          </ul>
          <p>Go to <a href="index.html">HOME</a> | <a href="document.html">Documentation</a> | <a href="fgrinn.html">fetchGrinnNetwork</a> | <a href="fcorrgrinn.html">fetchCorrGrinnNetwork</a> | <a href="fdiffcorr.html">fetchDiffCorrGrinnNetwork</a>
          | <a href="fmodule.html">fetchModuGrinnNetwork</a> | <a href="fgrinncorr.html">fetchGrinnCorrNetwork</a></p>
        <hr>
        <footer>
          &copy;KW UC Davis, 2015, Grinn is maintained by <a href="https://github.com/kwanjeeraw">kwanjeeraw</a><br>
          This page was generated by <a href="http://pages.github.com">GitHub Pages</a>. Tactile theme by <a href="https://twitter.com/jasonlong">Jason Long</a>.
        </footer>
      </section>
          <script type="text/javascript">
            var gaJsHost = (("https:" == document.location.protocol) ? "https://ssl." : "http://www.");
            document.write(unescape("%3Cscript src='" + gaJsHost + "google-analytics.com/ga.js' type='text/javascript'%3E%3C/script%3E"));
          </script>
          <script type="text/javascript">
            try {
              var pageTracker = _gat._getTracker("UA-61396514-2");
            pageTracker._trackPageview();
            } catch(err) {}
          </script>
  </body>
</html>
