<!DOCTYPE html>
<html lang="en-us">
  <head>
    <meta charset="UTF-8">
    <title>grinn R package</title>
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <link rel="stylesheet" type="text/css" href="stylesheets/normalize.css" media="screen">
    <link href='http://fonts.googleapis.com/css?family=Open+Sans:400,700' rel='stylesheet' type='text/css'>
    <link rel="stylesheet" type="text/css" href="stylesheets/stylesheet.css" media="screen">
    <link rel="stylesheet" type="text/css" href="stylesheets/github-light.css" media="screen">
  </head>
  <body>
    <section class="page-header">
      <img src="images/logo.png" alt="" width="450px">
      <a href="https://github.com/kwanjeeraw/grinn" class="btn">View on GitHub</a>
      <a href="https://github.com/kwanjeeraw/grinn/zipball/master" class="btn">Download .zip</a>
      <a href="https://github.com/kwanjeeraw/grinn/tarball/master" class="btn">Download .tar.gz</a>
    </section>
<section class="main-content">
      <hr>
          <h2><a id="grinn" class="anchor" href="#grinn" aria-hidden="true"><span class="octicon octicon-link"></span></a>grinn</h2>
          <p>grinn is a bioinformatics platform containing an internal graph database (using Neo4j), and the R package for omics studies.
          grinn incorporates data from several databases including KEGG, SMPDB, HMDB, REACTOME and CheBI.
          We apply correlation-based analyses from widely used WGCNA and DiffCorr R package.</p>
          <p>The grinn package provides R functions for:</p>
          <ol>
            <li> Reconstruction of biological networks in different level e.g. metabolite-protein-gene, metabolite-protein, metabolite-pathway, protein-gene, protein-pathway and gene-pathway</li>
            <li> Correlation-based analysis independently from pathway information to estimate:
              <ul>
              <li style="list-style-type:square">relationships among multi-omic levels</li>
              <li style="list-style-type:square">differential correlations among multi-omic levels in 2 conditions</li>
              <li style="list-style-type:square">relationships between omics data and phenotypic data</li>
              </ul>
            </li>
            <li> Rapid integration of domain knowledges to unknowns</li>
          </ol>
          <h3><a id="tutorials" class="anchor" href="#tutorials" aria-hidden="true"><span class="octicon octicon-link"></span></a>Tutorials</h3>
          <p>The tutorials provides the examples of how to use different grinn functions for the analysis of omics data. Click <a href="extra/grinn_data.zip">here</a> for the datasets used in these tutorials.</p>
          <ol>
              <li><a href="fgrinn.html">fetchGrinnNetwork</a> - reconstruct a biological network (grinn network) using information from grinn internal database</li>
              <li><a href="fcorrgrinn.html">fetchCorrGrinnNetwork</a> - compute a weighted correlation network and expand the network with information from grinn internal database</li>
              <li><a href="fdiffcorr.html">fetchDiffCorrGrinnNetwork</a> - compute a differential correlation network and expand the network with information from grinn internal database</li>
              <li><a href="fmodule.html">fetchModuGrinnNetwork</a> - compute a network module correlated to a phenotypic feature and expand the network with information from grinn internal database</li>
              <li><a href="fgrinncorr.html">fetchGrinnCorrNetwork</a> - reconstruct a biological network (grinn network) using information from grinn internal database, then compute and combine with a weighted correlation network</li>
          </ol>
          <h3><a id="installation" class="anchor" href="#installation" aria-hidden="true"><span class="octicon octicon-link"></span></a>Installation</h3>
          <ol>
          <li>Require <a href="http://www.r-project.org/" target="_blank">R</a> software and dependent R packages including RCurl, jsonlite, igraph, WGCNA, Hmisc, plyr, stringr, reshape2.</li>
          <li>Download grinn R package, click <a href="extra/grinn_2.0.tgz">here</a></li>
            <pre><code>
#1. install dependent R packages, if not exist
install.packages(c("matrixStats", "Hmisc", "splines", "foreach", "doParallel", "reshape", "fastcluster", "dynamicTreeCut", "survival", "RCurl", "jsonlite", "igraph", "WGCNA", "plyr", "stringr", "reshape2") 
source("http://bioconductor.org/biocLite.R") 
biocLite(c("GO.db", "preprocessCore", "impute"))
#2. install grinn package
install.packages("grinn_2.0.tgz", repo = NULL)
            </code></pre>
          <li>Require <a href="http://neo4j.com/download/" target="_blank">neo4j-community-2.2.0</a>, download and unzip.</li>
          <li>grinn internal database is available for local use. Please send us an email for the database file.</li>
          <li>Extract and move the database file to neo4j directory</li>
          <li>Start neo4j server, for windows: Double-click on %NEO4J_HOME%\bin\Neo4j.bat, for linux: ./bin/neo4j start. for more details see <a href="http://neo4j.com/docs/stable/server-installation.html" target="_blank">here</a></li>
          </ol>
          <h3><a id="references" class="anchor" href="#references" aria-hidden="true"><span class="octicon octicon-link"></span></a>References</h3>
          <p></p>
          <ul>
            <li>Croft D, <em>et al.</em> The Reactome pathway knowledgebase. <a href="http://nar.oxfordjournals.org/content/42/D1/D472.abstract" target="_blank">Nucleic Acids Res. 2014 Jan;42(Database issue):D472-7</a>.</li>
            <li>Degtyarenko K, <em>et al.</em> ChEBI: a database and ontology for chemical entities of biological interest. <a href="http://nar.oxfordjournals.org/content/36/suppl_1/D344.abstract" target="_blank">Nucleic Acids Res 2008. 36, D344-D350</a>.</li>
            <li>Jewison T, <em>et al.</em> SMPDB 2.0: Big Improvements to the Small Molecule Pathway Database. <a href="http://nar.oxfordjournals.org/content/42/D1/D478.abstract" target="_blank">Nucleic Acids Res 2014 Jan;42(Database issue):D478-84</a>.</li>
            <li>Kanehisa M, <em>et al.</em> Data, information, knowledge and principle: back to metabolism in KEGG. <a href="http://nar.oxfordjournals.org/content/42/D1/D199.abstract" target="_blank">Nucleic Acids Res 2014. 42, D199-D205</a>.</li>
            <li>Wishart DS, <em>et al.</em> HMDB 3.0 - The Human Metabolome Database in 2013. <a href="http://nar.oxfordjournals.org/content/41/D1/D801.abstract" target="_blank">Nucleic Acids Res 2013. Jan 1;41(D1):D801-7</a>.</li>
          </ul>
          <p>Correlation-based analyses apply methods from the following publications:</p>
          <ul>
            <li>Dudoit S, <em>et al.</em> Statistical methods for identifying differentially expressed genes in replicated cDNA microarray experiments, <a href="http://citeseerx.ist.psu.edu/viewdoc/summary?doi=10.1.1.117.9702" target="_blank">STATISTICA SINICA 2002;12:111</a>.</li>
            <li>Fukushima A DiffCorr: an R package to analyze and visualize differential correlations in biological networks. <a href="http://www.sciencedirect.com/science/article/pii/S0378111912014497" target="_blank">Gene 2013;10;518(1):209-14</a>.</li>
            <li>Langfelder P, <em>et al.</em> WGCNA: an R package for weighted correlation network analysis. <a href="http://www.biomedcentral.com/1471-2105/9/559" target="_blank">BMC Bioinformatics 2008;9:559</a>.</li>
            <li>Langfelder P, <em>et al.</em> Tutorials for the WGCNA package, <a href="http://labs.genetics.ucla.edu/horvath/CoexpressionNetwork/Rpackages/WGCNA/Tutorials/index.html" target="_blank"> click here</a></li> 
          </ul>
          <p>Omics data used in our tutorials are taken from the following publications:</p>
          <ul>
            <!--li>April W, <em>et al.</em> Metabolomics in psoriatic disease: pilot study reveals metabolite differences in psoriasis and psoriatic arthritis. <a href="http://f1000research.com/articles/10.12688/f1000research.4709.1/doi" target="_blank">F1000Res. 2014;3:248</a>.</li-->
            <!--li>Rosenberg A, <em>et al.</em> Divergent gene activation in peripheral blood and tissues of patients with rheumatoid arthritis, psoriatic arthritis and psoriasis following infliximab therapy. <a href="http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0110657" target="_blank">PLoS One. 2014;9(10)</a>.</li-->
            <li>Wikoff WR, <em>et al.</em> Metabolomic markers of altered nucleotide metabolism in early stage adenocarcinoma. <a href="http://cancerpreventionresearch.aacrjournals.org/cgi/pmidlookup?view=long&pmid=25657018" target="_blank">Cancer Prev Res (Phila) 2015;8(5):410-8</a>.</li>
            <li>Zakaria N, <em>et al.</em> Human non-small cell lung cancer expresses putative cancer stem cell markers and exhibits the transcriptomic profile of multipotent cells. <a href="http://www.biomedcentral.com/1471-2407/15/84" target="_blank">BMC Cancer 2015;15:84</a>.</li>
          </ul>
          <!--<h3><a id="citation" class="anchor" href="#citation" aria-hidden="true"><span class="octicon octicon-link"></span></a>Citation</h3>
          <p><a href="#"></a></p>-->
        <hr>
        <footer>
          &copy;KW UC Davis, 2015, Grinn is maintained by <a href="https://github.com/kwanjeeraw">kwanjeeraw</a><br>
          This page was generated by <a href="http://pages.github.com">GitHub Pages</a>. Tactile theme by <a href="https://twitter.com/jasonlong">Jason Long</a>.
        </footer>
        </section>
          <script type="text/javascript">
            var gaJsHost = (("https:" == document.location.protocol) ? "https://ssl." : "http://www.");
            document.write(unescape("%3Cscript src='" + gaJsHost + "google-analytics.com/ga.js' type='text/javascript'%3E%3C/script%3E"));
          </script>
          <script type="text/javascript">
            try {
              var pageTracker = _gat._getTracker("UA-61396514-2");
            pageTracker._trackPageview();
            } catch(err) {}
          </script>
  </body>
</html>
